pkg/kubelet/config/common.go

```diff
@@ -54,6 +54,10 @@ func generatePodName(name string, nodeName types.NodeName) string {
 }
 
 func applyDefaults(pod *api.Pod, source string, isFile bool, nodeName types.NodeName) error {
+	if pod.Name == "" {
+		return fmt.Errorf("name is needed for the pod")
+	}
+
```

Comment:
I'd suggest not to mix applying defaults with validation. Do we have another place where static pods are validated? If not, we should probably create it.